src/main/java/uk/gov/hmcts/reform/civil/service/InitiateGeneralApplicationServiceHelper.java

```diff
@@ -218,6 +211,33 @@ private String getApplicantPartyName(CaseAssignedUserRolesResource userRoles, Us
         return EMPTY;
     }
 
+    public boolean isGAApplicantSameAsParentCaseClaimant(CaseData caseData, UserDetails userDetails) {
```

Comment:
I think this method can be broken down into small methods to avoid part of code repetition in this method and setRespondentDetailsIfPresent. Small method can be like "getApplicantSolictor", "getRespondentSolicitor", "getUserRole"